test/integration/loans.test.ts

```diff
@@ -137,18 +162,420 @@ describe('Loans', () => {
+          
+          const loanId =  helpers.details.loan.id 
+          console.log('loanId',loanId)
+
+        //  const lHelpers = await loanHelpers(loanId)  
+          
+
+
+          //need to give fake ERC20 to borrower 
+          //and add fake ERC20 to the borrow escrow 
+
+          const lendingToken =
+          typeof market.lendingToken === 'string' ? await tokens.get(market.lendingToken) : market.lendingToken
+
+          const lendingTokenDecimals = await lendingToken.decimals()
+          console.log('decimals', lendingTokenDecimals)
+
+          await getFunds({
+            to: borrower,
+            tokenSym: market.lendingToken,
+            amount: 200 * 10**(lendingTokenDecimals),
+            hre,
+          })
+
+          const borrowerAddress = await borrower.getAddress()
+          const borrowerBalance = await lendingToken.balanceOf(borrowerAddress) 
+
+          console.log('balanceOf', borrowerBalance.toString() )
+
+          console.log('market.lendingToken',market.lendingToken)
+
+
+          const balanceLeftToRepay =  await diamond.getTotalOwed( loanId )
+          console.log('balanceLeftToRepay',balanceLeftToRepay.toString())
+
+
+
+          
+
+          await lendingToken
+          .connect(ethers.provider.getSigner(borrowerAddress))
+          .approve(diamond.address, balanceLeftToRepay)
+
+         
+          await diamond.connect( borrower ).repayLoan(loanId, 100010000)
+
+          
+          const totalOwedAfterRepay = await diamond.getTotalOwed( loanId )
+
+
+          console.log('totalOwedAfterRepay',  totalOwedAfterRepay.toString()  )
+
+
+          const loanData = await diamond.getLoan(loanId)
+          console.log('loanData',  ) 
+
+         
+          console.log('balanceLeftToRepay 2',balanceLeftToRepay.toString())
+
+          console.log('balanceOf After', borrowerBalance.toString() )
+
+          expect(parseInt(borrowerBalance.toString())).to.equal(499980000)
+
+           
+          expect(loanData.status  ).to.equal(3) //3 = repaid 
+
+          expect(parseInt( totalOwedAfterRepay.toString() ) ).to.equal(0) 
         })
+
+
+
+        it('should be able to create, earn profits on, and repay a loan with excess profit ', async () => {
+
+
+          const { getHelpers } = await takeOutLoanWithNfts({
+            amount: 100,  //should use raw amt not formatted.. oh well 
+            lendToken: market.lendingToken,
+          })
+
+          const helpers: LoanHelpersReturn = await getHelpers()
+
+
+          expect(helpers.details.loan).to.exist
+
+          console.log('helpers.details.loan', helpers.details.loan)
+       
+
+
+         // console.log('helpers.details.loan', helpers.details.loan)
+          
+          const loanId =  helpers.details.loan.id 
+          console.log('loanId',loanId)
+
+        //  const lHelpers = await loanHelpers(loanId)  
+          
+
+
+          //need to give fake ERC20 to borrower 
+          //and add fake ERC20 to the borrow escrow 
```

Comment:
is this to test profits with a different token? in order to calculate profits in the lending token, this token would have to have a chainlink aggregator registered via the ENS registry